core/src/test/scala/unit/kafka/cluster/PartitionTest.scala

```diff
@@ -945,7 +945,7 @@ class PartitionTest extends AbstractPartitionTest {
       // Invoke some operation that acquires leaderIsrUpdate write lock on one thread
       executor.submit((() => {
         while (!done.get) {
-          partitions.foreach(_.maybeShrinkIsr())
+          partitions.foreach(_.removeFutureLocalReplica(false))
```

Comment:
This was being used for acquiring the write-lock, which it won't do now. removeFutureLocalReplica is a no-op, but it may be worth exposing the lock to tests.